tooling/cli/src/dev.rs

```diff
@@ -345,14 +341,17 @@ pub fn setup(
         dev_url = Some(server_url.parse().unwrap());
 
         if let Some(c) = &options.config {
-          let mut c: tauri_utils::config::Config = serde_json::from_str(c)?;
+          let mut c: tauri_utils::config::Config = serde_json::from_value(c.0.clone())?;
           c.build.dev_url = dev_url.clone();
-          options.config = Some(serde_json::to_string(&c).unwrap());
+          options.config = Some(crate::ConfigValue(serde_json::to_value(&c).unwrap()));
         } else {
-          options.config = Some(format!(r#"{{ "build": {{ "devUrl": "{server_url}" }} }}"#))
+          options.config = Some(crate::ConfigValue(
+            serde_json::from_str(&format!(r#"{{ "build": {{ "devUrl": "{server_url}" }} }}"#))
+              .unwrap(),
+          ))
         }
```

Comment:
We should probably refactor this to change the value in-place without deserializing and serializing again.

Reply:
good idea, done